pkg/api/login_oauth.go

```diff
@@ -199,6 +199,12 @@ func (hs *HTTPServer) OAuthLogin(ctx *m.ReqContext) {
 				if org.OrgID == 0 {
 					org.OrgID = 1
 				}
+
+				if !isRoleAssignable(extUser.OrgRoles[org.OrgID], m.RoleType(org.Role)) {
+					continue
```

Comment:
Not really sure where to look, but I assume an empty `OrgRoles` map is okay?

Reply:
Yes.  I also added a test for this.